source/common/router/config_impl.cc

```diff
@@ -480,6 +480,43 @@ void RouteEntryImplBase::finalizePathHeader(Http::HeaderMap& headers,
   headers.Path()->value(path.replace(0, matched_path.size(), rewrite));
 }
 
+absl::string_view RouteEntryImplBase::ProcessRequestHost(const Http::HeaderMap& headers,
+                                                         const absl::string_view& new_scheme,
+                                                         const absl::string_view& new_port) const {
+
+  absl::string_view request_host = headers.Host()->value().c_str();
+  size_t host_end;
+  // Detect if IPv6 URI
+  if (request_host[0] == '[') {
+    host_end = request_host.rfind("]:");
+    if (host_end != absl::string_view::npos) {
+      host_end += 1; // advance to :
+    }
+  } else {
+    host_end = request_host.rfind(":");
+  }
+
+  if (host_end != absl::string_view::npos) {
+
+    absl::string_view request_port = request_host.substr(host_end);
+    absl::string_view request_scheme = headers.ForwardedProto()->value().c_str();
```

Comment:
Should this be renamed to indicate it's more about the transport than the scheme? 

If a client send a request  http://foo.com:443/bar and we have a redirect rule for https redirect, I believe we ought to redirect to https://foo.com[no 443 necessary].  In this case I think your code works since it's based on ForwardedProto (which would be https in this case) but the actual scheme of the request url would be http.

Reply:
I'm not sure about that. I mean we're calling it a "scheme" on all other layers starting from `route.proto` where we defined `scheme_redirect`.
I don't have a strong opinion here and can rename it, if that's what is needed.

Reply:
I think either I was unsufficiently clear here, or our implementation of x-forwarded-proto is incorrect :-)  If x-forwarded-proto is supposed to indicate scheme, not client-proxy protocol, we're doing it wrong, because if we get a fully qualified http scheme request over TLS we'll insert an X-Forwarded-Proto of https.

I assume from it being X-Forwarded-Proto rather than X-Forwarded-Scheme it's about the underlying protocol, and request_protocol is more accurate than request_scheme.  We're going to do a redirect to the https scheme(so scheme_redirect seems fine) but for port clearing we care about the underlying application protocol rather than scheme right?

Reply:
Thanks for the clarification @alyssawilk, it's probably just me misunderstanding your point. It makes sense now. Please let me know if you want me to change the variable now that you've approved the changes.